### ISSUE TITLE
problem with escaping commas handling multiple categories [sf#75]

### ISSUE THREAD

**Issue by winterz** [libical/libical#75]
A vCard may contain multiple categories.

CATEGORIES property of VCalender includes comma ( , ) separated Categories 

CATEGORIES:category1,category2

In the event that a category contains a comma this comma should be escaped.
For Example

CATEGORIES:this is\, category1,this is\, category2

there are following two functions available in libical 

1) icalproperty_new_categories(const char * )

This function can be used in case there is only one Category and will replace commas(,) with \,

2) icalproperty_vanew_categories(const char *, .... )

This function is supposed to take all the categories and create a comma separated string out of it and add this to Categories.
but this function is not working properly and it add only first parameter. 
In short this function is behaving in similar fashion as the first function.

Provided the above circumstances there is no way we can create a CATEGORIES: property containing the information of Multiple Categories.

**Comment by ksmurchison** [comment]
It appears to me that icalproperty_vanew_categories() is like the rest of the icalproperty_vanew_*() functions in that the arg list are parameters to be set on the property, NOT additional values for a multi-valued property.

For multi-valued properties (CATEGORIES, RESOURCES, RDATE, FREEBUSY), do we want icalproperty_new_*() to be able to accept a comma-separated list and split it into multiple properties, like icalparser does, or should we force apps to call icalproperty_new_*() multiple times for a multi-valued property?

**Comment by winterz** [comment]
my gut feeling is we should go with accepting comma-separated lists and split it into multiple props.
unless that turns out to be a total pain.
